src/shadowsocks_config.ts

```diff
@@ -338,3 +339,63 @@ export const SIP002_URI = {
     return `ss://${userInfo}@${uriHost}:${port.data}/${queryString}${hash}`;
   },
 };
+
+export interface OnlineConfig {
+  // URL endpoint to retrieve a Shadowsocks configuration.
+  readonly url: string;
+  // Any additional configuration (e.g. `certFp`, `httpMethod`, etc.) may be stored here.
+  readonly extra: {[key: string]: string};
```

Comment:
Let's explicitly list the options we support instead (cert fp for now?).

Reply:
Done in b3f2f19.